### PR TITLE
[PW-6221]Added change to update aria-label on playbackbutton toggle

### PR DESCRIPTION
**Pull request by kishorens**
Problem: The `aria-label` is not updated when the playback button is toggled.

Fix: Added change to update the aria-label on playback button toggle

### CODE REVIEW

File: src/ts/components/playbacktogglebutton.ts [hunk 1]
```diff
@@ -39,8 +39,10 @@ export class PlaybackToggleButton extends ToggleButton<ToggleButtonConfig> {
 
       if (player.isPlaying() || this.isPlayInitiated) {
         this.on();
+        this.getDomElement().attr('aria-label', i18n.performLocalization(i18n.getLocalizer('pause')));
       } else {
         this.off();
+        this.getDomElement().attr('aria-label', i18n.performLocalization(i18n.getLocalizer('play')));
```
Comment (ravloony) [inline src/ts/components/playbacktogglebutton.ts:45]: Suggestion: You could maybe split these two virtually identical lines to a private method?

**Review by ravloony — CHANGES_REQUESTED** [review]
I just had a small nitpick

File: src/ts/components/playbacktogglebutton.ts [hunk 2]
```diff
@@ -39,11 +39,17 @@ export class PlaybackToggleButton extends ToggleButton<ToggleButtonConfig> {
...
+        updateAriaLabel('pause');
       } else {
         this.off();
+        updateAriaLabel('play');
       }
     };
 
+    const updateAriaLabel = (label: string) => {
+      this.getDomElement().attr('aria-label', i18n.performLocalization(i18n.getLocalizer(label)));
+    };
```
Comment (stonko1994) [inline src/ts/components/playbacktogglebutton.ts:51]: Every component in our UI should in theory support the `ariaLabel`. Therefore this 'update' possibility should be moved to the `component.ts`.

Also, I think the `updateAriaLabel` function shouldn't hide the dependency to the localization feature. I would prefer that this function takes a `LocalizableText` as an input parameter.

**Review by stonko1994 — COMMENTED** [review]
- Please don't use internal links/issue numbers in this repo as it's open-source (PR title)
- I had a quick look at the `component.ts` and saw that there is an `ariaLabel` inside the `ComponentConfig` though it's not used/picked up when building the DOM element. This seems like a bug to me. IMO at DOM element creation the `ariaLabel` should be applied.
  - Can you please verify this and create an internal issue for this if needed?

File: src/ts/components/playbacktogglebutton.ts (hunk 2 above)
Comment (ravloony) [reply src/ts/components/playbacktogglebutton.ts:51]: @stonko1994 I had a stab at this here: https://github.com/bitmovin/bitmovin-player-ui/compare/feature/fix-pause-aria-label...feature/fix-pause-aria-label-2

Comment (stonko1994) [reply src/ts/components/playbacktogglebutton.ts:51]: I had a look at it but I'm not super happy with the changes to the `ToggleButtonConfig` 🤔 

I can see a problem that the `ToggleButtonConfig` extends from the `ButtonConfig` which again extends the `ComponentConfig`. That means that the `ToggleButtonConfig` now has 3 `areaLabel` related properties:

```ts
interface ToggleButtonConfig extends ButtonConfig {
  ariaLabel?: LocalizableText; // From the `ComponentConfig`
  onAriaLabel?: LocalizableText;
  offAriaLabel?: LocalizableText;
}
```

IMO that is now super confusing what to use when someone creates a custom component or initializes a `ToggleButton` with a custom configuration 🤔 

_+1: The mentioned bug from the review comment is still there_

Comment (ravloony) [reply src/ts/components/playbacktogglebutton.ts:51]: Fully agree, I wasn't sure what to do though, so I just left it to see what you'd say.

I can see two options:

```
interface ToggleButtonConfig extends ButtonConfig {
  ariaLabel?: LocalizableText; // From the `ComponentConfig`
  offAriaLabel?: LocalizableText;
}
```
and doing some funky type magic to replace the ariaLabel in the inherited class with something like

```
ariaLabel: { on?: LocalizableText, off?: LocalizableText }
```

Comment (stonko1994) [reply src/ts/components/playbacktogglebutton.ts:51]: Hm I'm not sure either 🤔 Both not ideal.

Do we really need the configuration option? What about manually managing it for the `ToggleButton` 🤔

Comment (ravloony) [reply src/ts/components/playbacktogglebutton.ts:51]: Discussion happened on Slack. All points resolved in #431

**Pull request merged**